### PR TITLE
Support for esp8266 (nodemcuv2)

### PR DESCRIPTION
**Pull request by jekkos**
Some minor changes I had to do to make it compile for ESP8266 in platformio. I still need to flash + test it.

### CODE REVIEW

**Comment by jekkos** [comment]
Tested it on a nodemcuv2, and works but together with #30

**Comment by ph1p** [comment]
Thank you! :) Could you please rebase to the latest changes of `main`? After that I will check and merge it.

File: src/main.cpp [hunk 1]
```diff
@@ -28,11 +35,13 @@ void setup()
...
   // wifi
   int attempts = 0;
+  #ifdef ESP32
   WiFi.setHostname(WIFI_HOSTNAME);
+  #endif
   WiFi.begin(WIFI_SSID, WIFI_PASSWORD);
   while (WiFi.status() != WL_CONNECTED && attempts < 7)
   {
-    delay(500);
+    delay(2000);
```
Comment (jekkos) [inline src/main.cpp:44]: this was needed to make wifi work

Comment (ph1p) [reply src/main.cpp:44]: Added it to the latest commit

**Comment by jekkos** [comment]
Thanks for merging all open PRs, I rebased with main and reflashed nodemcu, had to add one more switch for the weather to compile, it works now again on my device.

**Comment by jekkos** [comment]
It seems that latest changes in main are now breaking esp8266. Some functionality is used for interrupts specific to esp32 

```
  hw_timer_t *Screen_timer = timerBegin(0, 80, true);
  timerAttachInterrupt(Screen_timer, &onScreenTimer, true);
  timerAlarmWrite(Screen_timer, 200, true);
  timerAlarmEnable(Screen_timer);
```
Will try to figure out what is the equivalent for the older chip. @ph1p am I correct that wiring does not change when using SPI for screen updates? The code that sets the brightness now uses more efficient library that pushes the bits to the shift register (and with brightness value)?

**Pull request closed without merging**